pygmt/datasets/earth_relief.py

```diff
@@ -14,9 +17,9 @@
 @kwargs_to_strings(region="sequence")
 def load_earth_relief(
     resolution="01d",
-    region=None,
-    registration=None,
-    data_source="igpp",
+    region: Union[str, Sequence, None] = None,
+    registration: Literal["gridline", "pixel", None] = None,
```

Comment:
> I feel the current doc style (style 4 in the above table) is the best one.

I agree that the signature of No 1 is too complicated. The highlighting of No 3 and No 4 is good, as it is consistent with the rest of the documentation.
Is it required to always give a default value? I am wondering if something like this can be a bit misleading (even though it is also given like this in the signature):
![283153676-d904cd92-67f1-4306-a73a-775a79229b4a_cut_default](https://github.com/GenericMappingTools/pygmt/assets/94163266/ef97ebf6-79da-4a96-b65e-7dde2679fc70)

Reply:
> Is it required to always give a default value? 

We don't have to give a default value, but I feel a default value in the function strings is good so that users don't have to go back to the function signature to know the default.



> I am wondering if something like this can be a bit misleading (even though it is also given like this in the signature): ![283153676-d904cd92-67f1-4306-a73a-775a79229b4a_cut_default](https://private-user-images.githubusercontent.com/94163266/287848545-ef97ebf6-79da-4a96-b65e-7dde2679fc70.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3NDI4NzAsIm5iZiI6MTcwMTc0MjU3MCwicGF0aCI6Ii85NDE2MzI2Ni8yODc4NDg1NDUtZWY5N2ViZjYtNzlkYS00YTk2LWI2NWUtN2RkZTI2NzlmYzcwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA1VDAyMTYxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlYjI5NjcxNjkxNWY0Y2Y0MzczZjBjN2I3Y2Y2MTM0ZmQzZjgxYjZlMzhmMzE2Y2JiMTc3ZjQ3MDUwNjkwMzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.z8bf-A8BFcCs5hjv8D52lvxX44ljJSh_dxWXWpJLad0)

Yes, it's misleading. We can rephrase it to something like: 

> Default is `None`, which means `"gridline"` for all resolutions except ...

Reply:
> We don't have to give a default value, but I feel a default value in the function strings is good so that users don't have to go back to the function signature to know the default.

I also think it's good to have the default value in the function string. Just thought about not stating it for cases like above, but rephrasing the string should also prevent confusion.